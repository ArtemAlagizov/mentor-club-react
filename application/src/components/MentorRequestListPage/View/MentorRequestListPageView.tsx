import React from "react";
import translator from "../../../config/i18next";
import { Styled } from "../Styled";
import { Link } from "react-router-dom";
import { IMentorRequestListPageViewProps } from "./IMentorRequestListPageViewProps";
import { translateMentorRequestStatus } from "../../../service/MentorRequestStatusHelper/MentorRequestStatusHelper";
import { Alert, Spin } from "antd";

const MentorRequestListPageView: React.FC<IMentorRequestListPageViewProps> = ({
  handleRowClick,
  mentorRequests,
  isLoading,
  error
}) => {
  const StatusCell: React.FC<{ rowData: any }> = ({ rowData }) => {
    return <>{translateMentorRequestStatus(rowData.status)}</>;
  };

  return (
    <Styled.Container>
      <Styled.TopHeader>
        <h1>{translator.t("mentorRequestListPage.title")}</h1>
        <Link to="/mentor-requests/new">
          <button data-id="new-mentor-request-button">
            {translator.t("mentorRequestListPage.newMentorRequest")}
          </button>
        </Link>
      </Styled.TopHeader>

      {isLoading && (
        <Spin>
          <Styled.SpinnerBoundaries />
        </Spin>
      )}

      {error && (
        <Alert
          data-id="fetch-mentor-requests-error"
          type="error"
          message={error}
        />
      )}

      {mentorRequests && <div data-id="mentor-request-list-table"></div>}
    </Styled.Container>
  );
};

export default MentorRequestListPageView;
