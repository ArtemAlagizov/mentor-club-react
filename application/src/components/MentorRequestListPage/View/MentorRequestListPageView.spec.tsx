import React from "react";
import { shallow } from "enzyme";
import MentorRequestListPageView from "./MentorRequestListPageView";

describe("Given MentorRequestResultPageView", () => {
  describe("when rendering", () => {
    it("should have MentorRequestResultPageView matching snapshot", () => {
      const component = shallow(
        <MentorRequestListPageView
          isLoading={false}
          error={null}
          mentorRequests={[]}
          handleRowClick={jest.fn()}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
