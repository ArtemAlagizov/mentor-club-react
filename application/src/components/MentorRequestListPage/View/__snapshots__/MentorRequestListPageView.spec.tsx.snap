// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given MentorRequestResultPageView when rendering should have MentorRequestResultPageView matching snapshot 1`] = `
<styled.div>
  <styled.div>
    <h1>
      Mentor request list
    </h1>
    <Link
      to="/mentor-requests/new"
    >
      <button
        data-id="new-mentor-request-button"
      >
        New mentor request
      </button>
    </Link>
  </styled.div>
  <div
    data-id="mentor-request-list-table"
  />
</styled.div>
`;
